shared/common-adapters/input.desktop.js

```diff
@@ -64,145 +95,153 @@ class Input extends Component<void, Props, State> {
-    }
-    if (!this.props.small && this.props.floatingLabelText && this.props.hintText) {
-      // $FlowIssue this whole class needs to be cleaned up
-      hintStyle.top = 32
-    }
-
-    // HACK: We can't reset the text area style, so we need to counteract it by moving the wrapper up
-    const multilineStyleFix = {
-      height: 'auto',
-      position: 'relative',
-      // Other HACK: having a floating label affects position, but only in multiline
-      bottom: (this.props.floatingLabelText ? 30 : 6),
-      // tweak distance between entered text and floating label to match single-line
-      marginTop: 1,
-      // tweak distance between entered text and underline to match single-line
-      marginBottom: -2,
-    }
-    const inputStyle = this.props.multiline ? multilineStyleFix : {height: 'auto', top: 3}
-    const alignStyle = this.props.style && this.props.style.textAlign ? {textAlign: this.props.style.textAlign} : {textAlign: 'center'}
-
-    const passwordVisible = this.props.type === 'passwordVisible'
-    const password = this.props.type === 'password'
+    const underlineColor = this._underlineColor()
+    const defaultRowsToShow = Math.min(2, this.props.rowsMax || 2)
+    const containerStyle = this._containerStyle(underlineColor)
+
+    const commonInputStyle = {
+      ...globalStyles.fontSemibold,
+      fontSize: _headerTextStyle.fontSize,
+      lineHeight: `${_lineHeight}px`,
+      backgroundColor: globalColors.transparent,
+      flex: 1,
+      border: 'none',
+      outlineWidth: 0,
+      ...(this.props.small
+      ? {textAlign: 'left'}
+      : {
+        textAlign: 'center',
+        minWidth: 333,
+        borderBottom: `1px solid ${underlineColor}`,
+      }),
+    }
+
+    const inputStyle = {
+      ...commonInputStyle,
+      height: 28,
+    }
+
+    const textareaStyle = {
+      ...commonInputStyle,
+      height: 'initial',
+      resize: 'none',
+      wrap: 'off',
+      paddingTop: 0,
+      paddingBottom: 0,
+      minHeight: this._rowsToHeight(this.props.rowsMin || defaultRowsToShow),
+      ...(this.props.rowsMax
+        ? {maxHeight: this._rowsToHeight(this.props.rowsMax)}
+        : {overflowY: 'hidden'}),
+    }
+
+    const floatingHintText = !!this.state.value.length &&
+      (this.props.hasOwnProperty('floatingHintTextOverride')
+       ? this.props.floatingHintTextOverride
+       : this.props.hintText || ' ')
+
+    const commonProps = {
+      autoFocus: this.props.autoFocus,
+      onBlur: this._onBlur,
+      onChange: this._onChange,
+      onFocus: this._onFocus,
+      onKeyDown: this._onKeyDown,
+      placeholder: this.props.hintText,
+      ref: r => { this._input = r },
+      value: this.state.value,
+    }
+
+    const singlelineProps = {
+      ...commonProps,
+      style: {...inputStyle, ...this.props.inputStyle},
+      type: {
+        password: 'password',
+        text: 'text',
+        passwordVisible: 'text',
+      }[this.props.type || 'text'] || 'text',
```

Comment:
you're validating the type.. maybe a method.

validateType (t: string): string {
  switch (t) {
    case 'password':
    case 'text':
    case 'passwordVisible':
      return t
    default:
     return 'text'
  }
}